src/neuroagent/app/dependencies.py

```diff
@@ -393,7 +394,68 @@ async def get_agent_memory(
         yield None
 
 
+def thread_to_vp(
+    user_id: Annotated[str, Depends(get_user_id)],
+    session: Annotated[Session, Depends(get_session)],
+    request: Request,
+    settings: Annotated[Settings, Depends(get_settings)],
+) -> dict[str, str]:
+    """From the current thread, get the corresponding vlab and project."""
+    if "x-project-id" in request.headers and "x-virtual-lab-id" in request.headers:
+        return {
+            "vlab_id": request.headers["x-virtual-lab-id"],
+            "project_id": request.headers["x-project-id"],
+        }
+    elif settings.keycloak.validate_token:
+        return {
+            "vlab_id": "430108e9-a81d-4b13-b7b6-afca00195908",
+            "project_id": "eff09ea1-be16-47f0-91b6-52a3ea3ee575",
+        }
+    else:
+        thread_id = str(request.url).split("/")[-1]
+        query = (
+            select(Threads)
+            .where(Threads.user_sub == user_id)
+            .where(Threads.thread_id == thread_id)
+        )
+        result = session.execute(query).all()
+        if result:
+            thread_info = result[0][0].__dict__
+        else:
+            raise HTTPException(
+                status_code=404,
+                detail="thread not found when trying to validate project ID.",
+            )
+        return {
+            "vlab_id": thread_info["vlab_id"],
+            "project_id": thread_info["project_id"],
+        }
+
+
+async def validate_project(
+    httpx_client: Annotated[AsyncClient, Depends(get_httpx_client)],
+    thread_to_vp: Annotated[dict[str, str], Depends(thread_to_vp)],
+    token: Annotated[str, Depends(get_kg_token)],
+    settings: Annotated[Settings, Depends(get_settings)],
+) -> str:
+    """Check user appartenance to vlab and project before running agent."""
+    response = await httpx_client.get(
+        f'{settings.virtual_lab.get_project_url}/{thread_to_vp["vlab_id"]}/projects/{thread_to_vp["project_id"]}',
+        headers={"Authorization": f"Bearer {token}"},
+    )
+
+    # There is a lot of information about the project in this response.
+    if response.status_code != 200:
+        raise HTTPException(
+            status_code=HTTP_401_UNAUTHORIZED,
+            detail="User does not belong to the project.",
+        )
+    else:
+        return thread_to_vp["project_id"]
```

Comment:
I see now that the function `validate_project` would be used in the `create_threads` function. So another proposition would be to not make it a dependency, rather just a normal function. Then we call this normal function in `thread_to_vp` (which is a dependency) and inside the `create_thread` endpoint. What do you think ? If we don't want to mix dependencies and non dependencies, it could be moved to some utils file if needed.